test/python/test_eval_json.py

```diff
@@ -0,0 +1,232 @@
+            "--factor=0.3",
+        ]
+    )
+    assert "Benchmarks that have improved:" in ret.stdout
+    assert "Benchmarks that have stayed the same:" in ret.stdout
+    assert "Benchmarks that have worsened:" in ret.stdout
+    assert ret.success
+
+
+@pytest.mark.script_launch_mode("subprocess")
+def test_cli_with_only_changed(script_runner: ScriptRunner) -> None:
+    """Testing the command line functionality with different parameters."""
+    ret = script_runner.run(
+        [
+            "compare",
+            "./test/python/results_baseline.json",
+            "./test/python/results_feature.json",
+            "--factor=0.2",
+            "--only_changed=true",
+        ]
+    )
+    assert "Benchmarks that have improved:" in ret.stdout
+    assert "Benchmarks that have stayed the same:" not in ret.stdout
+    assert "Benchmarks that have worsened:" in ret.stdout
+    assert ret.success
+
+
+@pytest.mark.script_launch_mode("subprocess")
+def test_cli_with_only_changed_and_no_split(script_runner: ScriptRunner) -> None:
+    """Testing the command line functionality with different parameters."""
+    ret = script_runner.run(
+        [
+            "compare",
+            "./test/python/results_baseline.json",
+            "./test/python/results_feature.json",
+            "--only_changed=true",
+            "--no_split=true",
+        ]
+    )
+    assert "All changed benchmarks:" in ret.stdout
+    assert "All benchmarks:" not in ret.stdout
+    assert "Benchmarks that have improved:" not in ret.stdout
+    assert "Benchmarks that have stayed the same:" not in ret.stdout
+    assert "Benchmarks that have worsened:" not in ret.stdout
+    assert ret.success
+
+
+@pytest.mark.script_launch_mode("subprocess")
+def test_cli_with_no_split(script_runner: ScriptRunner) -> None:
+    """Testing the command line functionality with different parameters."""
+    ret = script_runner.run(
+        [
+            "compare",
+            "./test/python/results_baseline.json",
+            "./test/python/results_feature.json",
+            "--no_split=true",
+        ]
+    )
+    assert "All benchmarks:" in ret.stdout
+    assert "All changed benchmarks:" not in ret.stdout
+    assert "Benchmarks that have improved:" not in ret.stdout
+    assert "Benchmarks that have stayed the same:" not in ret.stdout
+    assert "Benchmarks that have worsened:" not in ret.stdout
+    assert ret.success
+
+
+@pytest.mark.script_launch_mode("subprocess")
+def test_cli_with_sort_by_algorithm(script_runner: ScriptRunner) -> None:
+    """Testing the command line functionality with different parameters."""
+    ret = script_runner.run(
+        [
+            "compare",
+            "./test/python/results_baseline.json",
+            "./test/python/results_feature.json",
+            "--sort=algorithm",
+            "--only_changed=true",
+            "--no_split=true",
+        ]
+    )
+    assert "All changed benchmarks:" in ret.stdout
+    assert ret.success
```

Comment:
Why do all of these functions have the same docstring? 🙃  at that point the docstring does not add any value.